File: treamap/src/graphviz/convert.py
import json

def json_to_graphviz(json_data):
    # Extract information from JSON
    node = json_data.get("node")
    parent = json_data.get("parent")
    views = json_data.get("views", 0)
    sales = json_data.get("sales", 0)
    he_success_rate = json_data.get("he_success_rate", 0.0)
    
    # Format the high efficiency success rate as percentage
    he_success_rate_percent = f"{he_success_rate * 100:.2f}%"
    
    # Graphviz representation
    graphviz = f"""
graph {{
    fontname="Helvetica"
    node [shape=record, fontname="Helvetica"];
    rankdir=TB;
    splines=false;
    
    "{parent}" [label=<{{
        <B>{parent}</B>|{{|}}|{{--%}}
    }}>, style=filled, fillcolor="#d2d1d1", color="white", fontcolor="white"];
    "{node}" [label=<{{
        <B>{node}</B>|{{{views}|{sales}}}|{{he_success_rate_percent}}}
    }}>;
    
    "{parent}" -- "{node}"
}}
"""
    return graphviz

# Load JSON data from a file
with open('sample.json', 'r') as file:
    json_data = json.load(file)

# Convert JSON to Graphviz format
graphviz_output = json_to_graphviz(json_data)
print(graphviz_output)

File: treamap/src/graphviz/convert.js
const fs = require('fs');

function jsonToGraphviz(filePath) {
  // Read JSON file
  fs.readFile(filePath, 'utf8', (err, data) => {
    if (err) {
      console.error('Error reading the file:', err);
      return;
    }

    // Parse the JSON data
    const jsonData = JSON.parse(data);
    let graphviz = `
graph {
    fontname="Helvetica"
    node [shape=record, fontname="Helvetica"];
    rankdir=TB;
    splines=false;
`;

    // Process each node
    jsonData.forEach(nodeData => {
      const { node, parent, views = 0, sales = 0, he_success_rate = 0.0, color } = nodeData;
      const invisible = false;
      const heSuccessRatePercent = (he_success_rate * 100).toFixed(2) + '%';
      const fillColor = color == 'red' ? '#FF8884' : color == 'blue' ? '#8bc34a' : 'silver';

      // Append to the graphviz string
      graphviz += `
    "${node}" [label=<{<B>${node}</B>|{${views}|${sales}}|{${heSuccessRatePercent}}}>, style=${invisible ? 'invis': 'filled'}, fillcolor="${fillColor}"];
    "${parent}" -- "${node}"${invisible ? ' [style=invis]' : ''};
`;
    });

    // Close the graph string
    graphviz += '}\n';

    console.log(graphviz);
  });
}

// Replace 'sample.json' with your actual file path
jsonToGraphviz('sample.json');
